Slm send datamode closure (#14561)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Serial LTE modem
----------------

> * Added:
> 
>   * The :ref:`CONFIG_SLM_DATAMODE_TERMINATOR <CONFIG_SLM_DATAMODE_TERMINATOR>` string is sent, if the connection with datamode is closed unexpectedly.

Reviewer: ```suggestion
* Changed:

  * The :ref:`CONFIG_SLM_DATAMODE_TERMINATOR <CONFIG_SLM_DATAMODE_TERMINATOR>` string to also be sent if the connection with data mode is closed unexpectedly.
```
I think this works better under being chagned, since the behavior (to my understanding) was already there when the connection was closed intentionally.

Author: The connection string was previously sent by the user to exit datamode. Now the application will also send the datamode, if the data pipe is broken. The user still needs to send the string to terminate the connection.
